numba/cuda/compiler.py

```diff
@@ -130,6 +132,35 @@ def compile_ptx_for_current_device(pyfunc, args, debug=False, device=False,
                        fastmath=fastmath, cc=cc, opt=True)
 
 
+def disassemble_cubin(cubin):
+    # nvdisasm only accepts input from a file, so we need to write out to a
+    # temp file and clean up afterwards.
+    fd = None
+    fname = None
+    try:
+        fd, fname = tempfile.mkstemp()
```

Comment:
Using `NamedTemporaryFile` here would simplify the code and make it less error prone. 
For reference: https://github.com/numba/numba/blob/master/numba/misc/inspection.py#L20-L22

Reply:
I can't see how to use `NamedTemporaryFile` here in a way that would simplify the logic or make it less error-prone: [its documentation](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) states that on Windows, the file can't be opened a second time whilst it is still open (as would be needed to open it with `nvdisasm`). So I'd have to close the file, open it with nvdisasm, and then finally delete it anyway.

- Am I missing something? How should the simpler version with `NamedTemporaryFile` work?
- If I'm not missing something, does the code you referenced suffer from the same problem? I had a look for tests that would hit it, but I couldn't find any.

Reply:
Good catch on the can't-open-file-twice limitation. I wasn't aware of that. And for the `disassemble_elf_to_cfg`, I don't think anyone has tested it on windows.